scripts/gz3d/build_parent_sample.py

```diff
@@ -0,0 +1,172 @@
+    for link in tqdm(links, desc="Downloading files"):
+        try:
+            download_file(base_url, link, download_dir=dir, force=force)
+        except Exception as e:
+            print(f"Error downloading {urljoin(base_url, link)}: {e}")
+        if test:
+            break
+
+def download_file(url: str, file_name: str, download_dir: str ='./data', chunk_size: int = 8192, force: bool = False):
+    """Download a file from a URL to a local directory in chunks (streaming).
+
+    Args:
+        url (str): URL of the file to download.
+        file_name (str): Name of the file to download.
+        download_dir (str, optional): Path to which the data is downloaded. Defaults to './data'.
+        chunk_size (int, optional): Chunk size for streaming. Defaults to 8192.
+
+    Returns:
+        str: Path to the downloaded file.
+    """
+
+    full_url = urljoin(url, file_name)
+
+    local_filename = download_dir / pathlib.Path(file_name)
+    if local_filename.exists() and not force:
+        return str(local_filename)
+    local_filename.parent.mkdir(parents=True, exist_ok=True)
+
+    with requests.get(full_url, stream=True) as r:
+        r.raise_for_status()
+        with local_filename.open('wb') as f:
+            for chunk in r.iter_content(chunk_size=chunk_size):
+                if chunk: # Skip empty chunks
+                    f.write(chunk)
+    
+    return str(local_filename)  # Return the file path as a string
+
+
+def _read_fits_file(loc):
+    with fits.open(loc) as hdul:
+        entry = pd.DataFrame(fits.open(loc)[5].data) 
+        entry.columns = map(str.lower, entry.columns) # lowercase column names
+        entry = Table.from_pandas(entry)
+
+        entry['false_color'] = hdul[0].data.astype(np.uint8)[None, ...]
+        entry['center'] = hdul[1].data.astype(np.uint8)[None, ...]
+        entry['star'] = hdul[2].data.astype(np.uint8)[None, ...]
+        entry['spiral'] = hdul[3].data.astype(np.uint8)[None, ...]
+        entry['bar'] = hdul[4].data.astype(np.uint8)[None, ...]
+    return entry
+
+def main(args):
+    # Download all files
+    if args.tiny:
+        limit = 5
+    elif args.limit:
+        limit = args.limit
+    else:
+        limit = None
+    
+    # Create the output directory
+    os.makedirs(args.output_dir, exist_ok=True)
+
+    # Download the data if it doesn't exist
+    if args.local_data_path == '':
+        files = build_filelist(GZ3D_URL, limit=limit)
+        download_all_files(files, GZ3D_URL, dir=args.output_dir, force=False)
+        local_data_path = args.output_dir
+    else:
+        local_data_path = args.local_data_path
+
+    # Read all files in parallel and aggregate the data into a single dataframe
+    locs = list(glob.glob(f"{local_data_path}/*.fits.gz"))
+
+    print(f"Found {len(locs)} FITS files")
+    with Pool() as pool:
+        entries = list(tqdm(pool.imap(_read_fits_file, locs), total=len(locs), desc="Reading FITS files"))
```

Comment:
`gz3d` fields are vertically flipped relative to `legacysurvey` fields. This change would flip the `gz3d` fields to be consistent with the `legacysurvey` fields. Note: this uses the `_vertical_flip_fields` function defined in another suggestion.
```suggestion
        entries = list(tqdm(pool.imap(lambda loc: _vertical_flip_fields(_read_fits_file(loc)), locs), total=len(locs), desc="Reading FITS files"))
        # _vertical_flip_fields gives the fields the same orientation as `legacysurvey` fields
```